### PR TITLE
Editor component py classes

### PR DESCRIPTION
**Pull request by michleza**
## What does this PR do?
- introduces editor component classes
- introduces editor component property classes
- prototype for the Physx Dynamic Rigid Body component
- includes a new test file to validate the Physx Dynamic Rigid Body component class
- test file added to Physics tests suite

### CODE REVIEW

File: AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_component/EditorEntityBoolProperty.py
Comment (hultonha) [inline AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_component/EditorEntityBoolProperty.py:6]: nit: It looks like maybe the comment needs to be updated here if this is `IntegerProperty` and not `BoolProperty`? (or maybe the type name is incorrect? Should it be `EditorEntityBoolProperty`?)

File: AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_component/EditorEntityVector3Property.py
Comment (hultonha) [inline AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_component/EditorEntityVector3Property.py:6]: update: Looks like the typename for this should be changed to `EditorEntityVector3Property`

File: AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_component/PhysXDynamicRigidBodyComponent.py
Comment (hultonha) [inline AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_component/PhysXDynamicRigidBodyComponent.py:47]: nit: We likely want to remove this comment before merging

**Review by hultonha — COMMENTED** [review]
Looks like a helpful addition, I just have a few minor comments. Nice work! 👍

File: AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_component/EditorEntityBoolProperty.py
Comment (michleza) [reply AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_component/EditorEntityBoolProperty.py:6]: yep this is a typo

File: AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_component/EditorEntityVector3Property.py
Comment (michleza) [reply AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_component/EditorEntityVector3Property.py:6]: yep. I scrambled to get this up to a PR for others but made some copypaste errors along the way

Comment (hultonha) [reply AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_component/EditorEntityVector3Property.py:6]: No worries, let me know when things are updated and I can re-review 👍

File: AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_component/EditorEntityComponentProperty.py
Comment (FuzzyCarterAWS) [inline AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_component/EditorEntityComponentProperty.py:18]: Remove Clamped.

This is unused and is a forward vision for implementation. 

Having this here early, and defaulting to true, is not providing any value to the libraries at this time, especially when we haven't determined the pattern for how to handle clamping (EG: you'd have to pass in some sort of data object with the clamped bounds when true to be stored and managed on the child class).

File: AutomatedTesting/Gem/PythonTests/editor/EditorScripts/Component_PhysX_Dynamic Rigid_Body.py
Comment (FuzzyCarterAWS) [inline AutomatedTesting/Gem/PythonTests/editor/EditorScripts/Component_PhysX_Dynamic Rigid_Body.py:1]: This file name has a space in it.

`Component_PhysX_Dynamic Rigid_Body.py`

**Review by FuzzyCarterAWS — APPROVED** [review]
This looks great and will be ready to merge after fixing a couple small pieces of feedback from my perspective.

* Please go through and add a new line to the end of every python file.
* A space in a file name
* Remove clamping as the design for this initial setup did not intend to solve how to implement it.

File: AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_component/EditorEntityComponentProperty.py
Comment (michleza) [reply AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_component/EditorEntityComponentProperty.py]: removed the clamped parameter.

File: AutomatedTesting/Gem/PythonTests/editor/EditorScripts/Component_PhysX_Dynamic Rigid_Body.py
Comment (michleza) [reply AutomatedTesting/Gem/PythonTests/editor/EditorScripts/Component_PhysX_Dynamic Rigid_Body.py]: added an underscore to fix this filename

**Review by moraaar — APPROVED** [review]
LGTM. Just one comment about the location of the files (not blocking)

File: AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_component/PhysXDynamicRigidBodyComponent.py
Comment (FuzzyCarterAWS) [reply AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_component/PhysXDynamicRigidBodyComponent.py:18]: Good catch.

@michleza a sub-folder for all the property classes would help keep things organized and clean

Comment (michleza) [reply AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_component/PhysXDynamicRigidBodyComponent.py]: ok. moved these to subfolders to keep things a bit tidier

**Pull request merged**
